From: William Tu <u9012063@gmail.com>
Date: Fri, 11 Aug 2017 06:46:39 -0700
Subject: selftests: bpf: add check for ip XDP redirect
Patch-mainline: v4.14-rc1
Git-commit: 047dbb27ff470782c4bd3d574942e571e747463f
References: bsc#1109837

Kernel test robot reports error when running test_xdp_redirect.sh.
Check if ip tool supports xdpgeneric, if not, skip the test.

Signed-off-by: William Tu <u9012063@gmail.com>
Acked-by: Daniel Borkmann <daniel@iogearbox.net>
Cc: John Fastabend <john.fastabend@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 tools/testing/selftests/bpf/test_xdp_redirect.sh |    5 +++++
 1 file changed, 5 insertions(+)

--- a/tools/testing/selftests/bpf/test_xdp_redirect.sh
+++ b/tools/testing/selftests/bpf/test_xdp_redirect.sh
@@ -23,6 +23,11 @@ cleanup()
 	ip netns del ns2 2> /dev/null
 }
 
+ip link set dev lo xdpgeneric off 2>/dev/null > /dev/null
+if [ $? -ne 0 ];then
+	echo "selftests: [SKIP] Could not run test without the ip xdpgeneric support"
+	exit 0
+fi
 set -e
 
 ip netns add ns1
